src/main/groovy/com/vherasymenko/avro/decoder/core/AvroDecoderService.groovy

```diff
@@ -0,0 +1,25 @@
+package com.vherasymenko.avro.decoder.core
+
+import event.course_install.CourseInstall
+import org.apache.avro.generic.GenericDatumReader
+import org.apache.avro.generic.GenericRecord
+import org.apache.avro.io.DatumReader
+import org.apache.avro.io.DecoderFactory
+
+/**
+ * The avro decoder.
+ */
+class AvroDecoderService implements AvroDecoderPort {
+
+    @Override
+    void decodeEvent( String event ) {
+        def inputStream = new ByteArrayInputStream( event.bytes )
```

Comment:
An aside, but are we just fighting ourselves using the JSON encoding with this? I was under the impression it would be over Rabbit as a JSON structure like we have already. If we aren't getting that, maybe that is proof enough that we should embrace the binary solution instead? I'm not saying halt all work and go down the other path or anything, I think it's a valid use case. However this is showing me that I had more misconceptions on implementation than I thought. If we aren't going to have a JSON structure going over Rabbit anyway and we have to encode the structure to bytes instead of passing around an actual string, that may be something that will complicate the current journal setup anyway. There is no decoding done currently, as it's a String structure assumed.

I may be simply making myself look like an idiot, since technically the event.bytes is simply an encoded JSON, but my point is that people can currently ( and the journal can ) look at the event in Rabbit and know what it is. If it's brought down to bytes, then we're losing that functionality.